src/cascade/executor/estimate_location.py

```diff
@@ -55,17 +68,111 @@ def retrieve_data(execution_context, local_settings):
         local_settings.parent_location_id, local_settings.sex_id,
         local_settings.data_access.gbd_round_id, ages_df, with_hiv=local_settings.data_access.with_hiv)
 
+    # These are the draws as output of the parent location.
+    data.draws = None
+
+    # The parent can also supply integrands as a kind of prior.
+    # These will be shaped like input measurement data.
+    data.integrands = None
+
     return data
 
 
-def compute_location(input_data, execution_context, local_settings):
+def modify_input_data(input_data, local_settings):
+    ev_settings = local_settings.settings
+    # These are suitable for input to the fit.
+    if hasattr(ev_settings.eta, "data") and ev_settings.eta.data:
+        data_eta = float(ev_settings.eta.data)
+    else:
+        data_eta = None
 
-    parent_location_id = local_settings.parent_location_id
-    global_data_eta = local_settings.settings.eta
-    observations = bundle_to_observations(input_data.bundle, parent_location_id, global_data_eta)
+    input_data.observations = bundle_to_observations(
+        input_data.bundle,
+        local_settings.parent_location_id,
+        data_eta
+    )
+    input_data.observations = input_data.observations.drop(columns="sex_id")
+    # ev_settings.data_eta_by_integrand is a dummy in form.py.
+    MATHLOG.info(f"Ignoring data_eta_by_integrand")
+
+    sorted_locations = list(nx.lexicographical_topological_sort(input_data.locations))
+    parents = list()
+    names = list()
+    for l in sorted_locations:
+        parent = list(input_data.locations.predecessors(l))
+        if parent:
+            parents.append(parent[0])
+        else:
+            parents.append(nan)
+        names.append(input_data.locations.nodes[l]["location_name"])
+    input_data.locations_df = pd.DataFrame(dict(
+        location_id=sorted_locations,
+        parent_id=parents,
+        name=names,
+    ))
+    return input_data
+
+
+def compute_location(execution_context, local_settings, input_data, model):
+    """
 
-    return observations
+    Args:
+        execution_context:
+        input_data: These include observations and initial guess.
+        model (Model): A complete Model object.
 
+    Returns:
+        The fit.
+    """
 
-def save_outputs(computed_fit, execution_context, local_settings):
+    session = Session(
+        locations=input_data.locations_df,
+        parent_location=model.location_id,
+        filename="subsession.db"
```

Comment:
This is so very not safe. Can you address this during running under asyncio? The execution context should know how / where to save any files.

Reply:
Is there a reason to not have the session stuff things in a tempdir unless you specify otherwise? And then you'd just leave the `filename` parameter off unless you wanted it someplace permanent for debugging or what not. Actually that answers my later question about what "closing" a session means. It means cleaning up the temporary directory.